r/R/install-arrow.R

```diff
@@ -79,7 +80,8 @@ install_arrow <- function(nightly = FALSE,
     # On the M1, we can't use the usual autobrew, which pulls Intel dependencies
     apple_m1 <- grepl("arm-apple|aarch64.*darwin", R.Version()$platform)
     # On Rosetta, we have to build without JEMALLOC, so we also can't autobrew
-    if (on_rosetta()) {
+    rosetta <- identical(sysname, "darwin") && identical(system("sysctl -n sysctl.proc_translated", intern = TRUE), "1")
+    if (rosetta) {
```

Comment:
```suggestion
    rosetta <- on_rosetta()
    if (rosetta) {
```

Another PR implemented a function `on_rosetta()` that already does these checks, so we can simplify this bit here, and remove the creation of the `sysname` variable above.

Reply:
Hi @thisisnic 

I believe that for the package this is fine. I saw that the `on_rosetta()` function was defined in `r/R/arrow-package.R` and that should work.

However, I'm thinking of the case where the user just uses the `r/R/install-arrow.R` as a standalone script, as suggested [here](https://arrow.apache.org/docs/r/articles/install.html#using-install_arrow) (as it's the way I use, for example). In this way, the `on_rosetta()` function is not found, as the script is not self-contained anymore.

So my proposal would be to just change the `in_rosetta()` function from `r/R/arrow-package.R` to `r/R/install-arrow.R`, as this would not affect the former, but it would make the latter self-contained again.

Reply:
I don't see the standalone use being suggested, maybe we need to clarify the docs? As I understand it `install_arrow` is for use when you have the arrow package installed but want to do one of the following:
```
You have arrow installed and want to upgrade to a different version
You want to try to reinstall and fix issues with Linux C++ binaries
You want to install a development build
```
@thisisnic please correct me if my assessment on that is wrong?

Reply:
@assignUser It's here: https://arrow.apache.org/docs/r/articles/install.html?q=install_arrow#using-install_arrow

> Although this function is part of the arrow package, it is also available as a standalone script, so you can access it without first installing the package:

In the case of "You want to install a development build", folks may not already have arrow installed.  

On a different note, there have been other PRs in this area of the codebase merged recently, so this PR may need a rebase to see how the changes affect it, and what needs doing.

Reply:
> So my proposal would be to just change the on_rosetta() function from r/R/arrow-package.R to r/R/install-arrow.R, as this would not affect the former, but it would make the latter self-contained again.

This is a good catch, I'm working on a review, but overall I'm pro moving this function to `install-arrow.R` so that it remains self-contained. Thank you for catching it!

>  In this way, the on_rosetta() function is not found, as the script is not self-contained anymore.

We don't have to do this here, but we should at least make an issue that we should write a test that tests that the script remains self-contained. It would have caught this bug, for example!